### PR TITLE
Run Nuclide Node tests

### PR DESCRIPTION
**Pull request by aminya**
- Starting from this PR, the Nuclide Node tests will run. 
- Jest tests runner for Node/Atom is added. 
- Several bugs are fixed.

Atom Tests Status:

Now, the Node tests run without issues, however, the Atom tests do not start.

Atom tests:
```
pnpm install
pnpm prepare.nuclide
pnpm test.nuclide.atom
```

node-ipc which is inside ipcServer file is not compatible with Electron 6 - Node 12.
https://github.com/RIAEvangelist/node-ipc/issues/163

### CODE REVIEW

**Comment by aminya** [comment]
@UziTech When I use the jasmine3 runner, the tests for nuclide/nuclide-adb does not run. Could you investigate why this happens? The default test runner works.

```
pnpm install
pnpm install.nuclide
pnpm build.nuclide-adb
cd nuclide/nuclide-adb
pnpm test
```

**Comment by UziTech** [comment]
Seem like another issue with pnpm. When I use npm it work fine.

**Comment by aminya** [comment]
I tested this with npm there is no difference. Where did you use npm?

**Comment by UziTech** [comment]
I see what you are saying. atom-jasmine3-test-runner is meant to be used to test atom packages with `window:run-package-specs` in Atom. Running tests on the command line with atom-jasmine3-test-runner is tricky because the `atom --test` command starts a separate process that runs the script that actually runs the tests so `console.log` doesn't always output to the console that `atom --test` was run in.

For npm packages it would be better to run the tests with jasmine instead of atom-jasmine3-test-runner.

**Comment by UziTech** [comment]
The reason it was failing when the test was lagacy was because jasmine didn't find any jasmine 3 tests (because there weren't any). This should be fixed in https://github.com/UziTech/atom-jasmine3-test-runner/pull/166

File: nuclide/atom-test-runners/jasmine-runner.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = createRunner({
+  testPackages: process.env.APM_TEST_PACKAGES ? process.env.APM_TEST_PACKAGES.split(' ') : [],
+  suffix: "-test",
+  legacySuffix: "-test-v1",
+  testPaths: ["__tests__", "__atom_tests__"],
+  timeReporter: true,
+  silentInstallation: true,
+  // Extra Packages
+  specHelper: false
+})
```
Comment (UziTech) [inline nuclide/atom-test-runners/jasmine-runner.js:13]:
```suggestion
const { createRunner } = require("atom-jasmine3-test-runner")

const testPaths = ["__tests__", "__atom_tests__"].filter(dir => {
  // check if folder exists here
})

// https://github.com/UziTech/atom-jasmine3-test-runner#api
module.exports = createRunner({
  testPackages: process.env.APM_TEST_PACKAGES ? process.env.APM_TEST_PACKAGES.split(' ') : [],
  suffix: "-test",
  legacySuffix: "-test-v1",
  testPaths,
  timeReporter: true,
  silentInstallation: true,
  // Extra Packages
  specHelper: false
})
```

File: nuclide/atom-test-runners/jasmine-runner.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { createRunner } = require("atom-jasmine3-test-runner")
+
+const fs = require("fs")
+
+// https://github.com/UziTech/atom-jasmine3-test-runner#api
+module.exports = createRunner({
+  testPackages: process.env.APM_TEST_PACKAGES ? process.env.APM_TEST_PACKAGES.split(' ') : [],
+  suffix: "-test-v3",
+  legacySuffix: "-test",
+  testPaths: ["__tests__", "__atom_tests__"].filter(dir => !fs.existsSync(process.cwd(), dir)),
```
Comment (UziTech) [inline nuclide/atom-test-runners/jasmine-runner.js:10]: `!fs.existsSync` filters the paths that dont exist.

and [existsSync](https://nodejs.org/api/fs.html#fs_fs_existssync_path) only takes one parameter. This should be `dir => fs.existsSync(path.resolve(dir))`

Comment (aminya) [reply nuclide/atom-test-runners/jasmine-runner.js:10]: Thanks. I did not have enough focus while I wrote this.

**Comment by aminya** [comment]
The jest runner does not transform the `imports` for some reason.

**Comment by aminya** [comment]
OK. I changed the build scripts so we don't have to use transform.

Now, the Node tests run without issues, however, that atom tests do not start.

```
pnpm install
pnpm build.nuclide
pnpm install.nuclide-test
pnpm test.nuclide
```

**Comment by aminya** [comment]
Now, the Node tests run without issues, however, the Atom tests do not start.

```
pnpm install
pnpm build.nuclide
pnpm install.dist-nuclide
pnpm test.nuclide.atom
```

**Comment by aminya** [comment]
node-ipc which is inside ipcServer file is not compatible with Node 12.
https://github.com/RIAEvangelist/node-ipc/issues/163

**Pull request merged**